src/unity/python/turicreate/visualization/show.py

```diff
@@ -96,46 +96,302 @@ def show(x, y, xlabel="X", ylabel="Y", title=None):
     path_to_client = _get_client_app_path()
     title = _get_title(title)
     plt_ref = tc.extensions.plot_scatter(path_to_client, x, y, 
       xlabel, ylabel,title)
     return Plot(plt_ref)
 
 def categorical_heatmap(x, y, xlabel="X", ylabel="Y", title=None):
+    """
+    Plots the data in `x` on the X axis and the data in `y` on the Y axis
+    in a 2d categorical heatmap, and returns the resulting Plot object.
+    
+    The function supports SArrays of dtypes str.
+
+    Parameters
+    ----------
+    x : SArray
+      The data to plot on the X axis of the categorical heatmap. 
+      Must be string SArray
+    y : SArray
+      The data to plot on the Y axis of the categorical heatmap. 
+      Must be string SArray and must be the same length as `x`.
+    xlabel : str (optional)
+      The text label for the X axis. Defaults to "X".
+    ylabel : str (optional)
+      The text label for the Y axis. Defaults to "Y".
+    title : str (optional)
+      The title of the plot. Defaults to None. If the value is None, the title
+      will be "<xlabel> vs. <ylabel>". Otherwise, the string passed in as the
+      title will be used as the plot title.
+
+    Return Value
+    ------------
+    An object of Plot that is the categorical heatmap
+
+    Examples
+    --------
+    Make a categorical heatmap.
+
+    >>> x = turicreate.SArray(['1','2','3','4','5'])
+    >>> y = turicreate.SArray(['a','b','c','d','e'])
+    >>> catheat = turicreate.visualization.categorical_heatmap(x, y)
+    """
+    if (type(x) != tc.data_structures.sarray.SArray or 
+        type(y) != tc.data_structures.sarray.SArray or
+        x.dtype != str or y.dtype != str):
+        raise ValueError("turicreate.visualization.categorical_heatmap supports " + 
+            "SArrays of dtype: str")
+    # legit input
     path_to_client = _get_client_app_path()
     title = _get_title(title)
     plt_ref = tc.extensions.plot_categorical_heatmap(path_to_client, x, y, 
       xlabel, ylabel, title)
     return Plot(plt_ref)
 
 def heatmap(x, y, xlabel="X", ylabel="Y", title=None):
+    """
+    Plots the data in `x` on the X axis and the data in `y` on the Y axis
+    in a 2d heatmap, and returns the resulting Plot object.
+    
+    The function supports SArrays of dtypes int, float.
+
+    Parameters
+    ----------
+    x : SArray
+      The data to plot on the X axis of the heatmap. 
+      Must be numeric (int/float)
+    y : SArray
+      The data to plot on the Y axis of the heatmap. 
+      Must be numeric (int/float) and must be the same length as `x`.
+    xlabel : str (optional)
+      The text label for the X axis. Defaults to "X".
+    ylabel : str (optional)
+      The text label for the Y axis. Defaults to "Y".
+    title : str (optional)
+      The title of the plot. Defaults to None. If the value is None, the title
+      will be "<xlabel> vs. <ylabel>". Otherwise, the string passed in as the
+      title will be used as the plot title.
+
+    Return Value
```

Comment:
Same comment as for scatter plot (should be Returns, with name/type/description in special syntax).